addons/product/models/product_template.py

```diff
@@ -848,63 +862,81 @@ def _get_first_possible_combination(self, parent_combination=None, necessary_val
         :param necessary_values: values that must be in the returned combination
         :type necessary_values: recordset of `product.template.attribute.value`
 
-        :return: the first possible combination found, or empty if none possible
-        :rtype: recordset of `product.template.attribute.value`
+        :return: the possible combinations
+        :rtype: generator of recordset of `product.template.attribute.value`
         """
         self.ensure_one()
 
         if not self.active:
-            return self.env['product.template.attribute.value']
+            return _("The product template is archived so no combination is possible.")
```

Comment:
return Falsy value (False or empty browse_Record) instead of string
idem below

Reply:
Ok, I can just return nothing then. The idea was to give the user a better message if that exception is raised, but I guess we can consider it a dev error if that happens and not worry about it?

Reply:
as discuss, since string will be the exception message, => we keep it